include/boost/graph/rank_spanning_branchings.hpp

```diff
@@ -0,0 +1,1159 @@
+      else
+        return true;
+
+    }
+
+    // Retrieve the path back from leaf to root in cycle branching.
+
+    void
+    find_back_path(
+      BranchingGraph& C,
+      std::vector<BranchingVertex>& bv
+    )
+    {
+
+      BGL_FORALL_INEDGES( bv[0], e, C, BranchingGraph )
+      {
+        bv.insert( bv.begin(), source( e, C ) );
+        find_back_path( C, bv );
+      }
+
+    }
+
+    // Expand cycles.
+
+    template <typename Graph, typename Edge, typename IndexMap,
+              typename WeightMap, typename Compare>
+    void
+    expand(
+      const Graph& g,
+      IndexMap& v_id,
+      BranchingGraph& C,
+      unordered_set<BranchingVertex>& root_set,
+      std::vector<EdgeNode< Edge, WeightMap, Compare> >& beta
+    )
+    {
+
+      BOOST_FOREACH( BranchingVertex v, root_set )
+      {
+        if( in_degree( v, C ) == 0 && out_degree( v, C ) != 0 ) 
+        {
+          std::vector<BranchingVertex> bv;
+          bv.push_back( v_id[target( beta[v].edge, g )] );
+          find_back_path( C, bv );
+          for( std::size_t i = 0; i < bv.size() - 1; i++ )
+          {
+            beta[bv[i+1]] = beta[bv[i]];
+            clear_vertex( bv[i], C );
+          }
+        }
+      }
+
+      // Remove isolated vertices.
+
+      std::vector<BranchingVertex> vertices_to_remove_from_set;
+
+      BOOST_FOREACH( BranchingVertex v, root_set )
+      {
+        if( in_degree( v, C ) == 0 && out_degree( v, C ) == 0 ) 
+        {
+          vertices_to_remove_from_set.push_back( v );
+        }
+      }
+
+      BOOST_FOREACH( BranchingVertex v, vertices_to_remove_from_set )
+      {
+        root_set.erase( v );
+      }
+
+    }
+
+    // Camerini et al. BEST routine.
+
+    template <typename Graph, typename Edge, typename IndexMap,
+              typename WeightMap, typename Rank, typename Pred,
+              typename Compare>
+    void
+    best_spanning_branching( const Graph& g, 
+                             unordered_set<Edge>& branching,
+                             IndexMap& v_id,
+                             WeightMap& w,
+                             Compare& comp,
+                             Rank rank,
+                             Pred pred1,
+                             Pred pred2,
+                             unordered_set<Edge>& include_edges,
+                             unordered_set<Edge>& exclude_edges
+                           )
+    {
+
+      typedef typename graph_traits<Graph>::vertex_descriptor Vertex;
+      typedef typename graph_traits<Graph>::vertices_size_type vertex_idx_t;
+
+      typedef EdgeNode<Edge, WeightMap, Compare> edge_node_t;
+
+      typedef heap::fibonacci_heap<edge_node_t> f_heap_t;
+      typedef std::map<Vertex, edge_node_t > exit_map_t;
+
+      unordered_set<Vertex> unvisited_vertex_set;
+
+      Vertex root_vertex;
+
+      BranchingGraph C;
+
+      vertex_idx_t n = num_vertices(g);
+
+      disjoint_sets<Rank, Pred> W( rank, pred1 ), S( rank, pred2 );
```

Comment:
I understand the motivation for single-letter variable names like `W`, `S`, `C`, etc, but in my experience, it's better to use a descriptive name. Not everyone has the algorithm paper on hand to look up what the letters mean. And it makes the variables awfully hard to text search for in a file, such as in this review.
`g` for graph, `u` and `v` for vertices, etc I think is fine... there is no perfect rule about it.

Reply:
Thanks, I agree with this suggestion.  Going through and making the changes will probably only reinforce your point.